source/application/AVIDumping/AudioConverterStream.cpp

```diff
@@ -4,253 +4,285 @@
-	{
-		m_failed = true;
-	}
+    if (sourceFormat.GetSize() == destFormat.GetSize())
+    {
+        LPWAVEFORMATEX src = static_cast<LPWAVEFORMATEX>(sourceFormat);
+        LPWAVEFORMATEX dest = static_cast<LPWAVEFORMATEX>(destFormat);
+
+        if (memcmp(src, dest, sourceFormat.GetSize()) == 0)
+        {
+            // Source and destination formats are identical, so we'll just pass the input through to
+            // the output
+            isProxy = true;
+            return;
+        }
+    }
+
+    MMRESULT mm = acmStreamOpen(
+        &stream, nullptr, sourceFormat, destFormat, nullptr, 0, 0, ACM_STREAMOPENF_NONREALTIME);
+
+    if (mm != MMSYSERR_NOERROR)
+    {
+        // Not supported directly, so try letting ACM suggest an intermediate format, so we can
+        // perform the conversion in multiple stages.
+        WaveFormat intermediateFormat;
+
+        // This weird looping structure is because we must try all combinations of four flags, and
+        // even when a given combination yields a valid suggestion, we must be able to backtrack and
+        // continue looping (tryMoreSuggestions) if the conversion based on that suggestion later
+        // fails or dead-ends.
+        bool foundSuggest = false;
+        int chan, samp, bits, form, done;
+
+        for (done = 0; done < 1 && !foundSuggest; foundSuggest ? 0 : ++done)
+        {
+            for (chan = 1; chan >= 0 && !foundSuggest; foundSuggest ? 0 : --chan)
+            {
+                for (samp = 1; samp >= 0 && !foundSuggest; foundSuggest ? 0 : --samp)
+                {
+                    for (bits = 0; bits <= 1 && !foundSuggest; foundSuggest ? 0 : ++bits)
+                    {
+                        for (form = 0; form <= 1 && !foundSuggest; foundSuggest ? 0 : ++form)
+                        {
+                            {
+                                int flags = 0;
+                                flags |= chan ? ACM_FORMATSUGGESTF_NCHANNELS : 0;
+                                flags |= samp ? ACM_FORMATSUGGESTF_NSAMPLESPERSEC : 0;
+                                flags |= bits ? ACM_FORMATSUGGESTF_WBITSPERSAMPLE : 0;
+                                flags |= form ? ACM_FORMATSUGGESTF_WFORMATTAG : 0;
+
+                                intermediateFormat = destFormat;
+                                MMRESULT mmSuggest = acmFormatSuggest(NULL,
+                                                                      sourceFormat,
+                                                                      intermediateFormat,
+                                                                      intermediateFormat.GetSize(),
+                                                                      flags);
+                                if (mmSuggest == MMSYSERR_NOERROR)
+                                {
+                                    // Got a possibly-valid suggestion, but it might be a suggestion
+                                    // to
```

Comment:
Ouch. This got bad...

Reply:
Hmm, looks like it's due to the comments being manually broken beforehand.